src/interpreter/ByteCode.h

```diff
@@ -30,23 +28,234 @@
+    F(I64DivU, intDiv, uint64_t, uint64_t)        \
+    F(I64RemS, intRem, int64_t, int64_t)          \
+    F(I64RemU, intRem, uint64_t, uint64_t)        \
+    F(I64And, intAnd, int64_t, int64_t)           \
+    F(I64Or, intOr, int64_t, int64_t)             \
+    F(I64Xor, intXor, int64_t, int64_t)           \
+    F(I64Shl, intShl, int64_t, int64_t)           \
+    F(I64ShrS, intShr, int64_t, int64_t)          \
+    F(I64ShrU, intShr, uint64_t, uint64_t)        \
+    F(I64Rotl, intRotl, uint64_t, uint64_t)       \
+    F(I64Rotr, intRotr, uint64_t, uint64_t)       \
+    F(I64Eq, eq, int64_t, int32_t)                \
+    F(I64Ne, ne, int64_t, int32_t)                \
+    F(I64LtS, lt, int64_t, int32_t)               \
+    F(I64LtU, lt, uint64_t, uint32_t)             \
+    F(I64LeS, le, int64_t, int32_t)               \
+    F(I64LeU, le, uint64_t, uint32_t)             \
+    F(I64GtS, gt, int64_t, int32_t)               \
+    F(I64GtU, gt, uint64_t, uint32_t)             \
+    F(I64GeS, ge, int64_t, int32_t)               \
+    F(I64GeU, ge, uint64_t, uint32_t)             \
+    F(F64Add, add, double, double)                \
+    F(F64Sub, sub, double, double)                \
+    F(F64Mul, mul, double, double)                \
+    F(F64Div, floatDiv, double, double)           \
+    F(F64Max, floatMax, double, double)           \
+    F(F64Min, floatMin, double, double)           \
+    F(F64Copysign, floatCopysign, double, double) \
+    F(F64Eq, eq, double, int32_t)                 \
+    F(F64Ne, ne, double, int32_t)                 \
+    F(F64Lt, lt, double, int32_t)                 \
+    F(F64Le, le, double, int32_t)                 \
+    F(F64Gt, gt, double, int32_t)                 \
+    F(F64Ge, ge, double, int32_t)
+
+#define FOR_EACH_BYTECODE_UNARY_OP(F)           \
+    F(I32Clz, clz, uint32_t, uint32_t)          \
+    F(I32Ctz, ctz, uint32_t, uint32_t)          \
+    F(I32Popcnt, popCount, uint32_t, uint32_t)  \
+    F(I32Eqz, intEqz, uint32_t, uint32_t)       \
+    F(F32Sqrt, floatSqrt, float, float)         \
+    F(F32Ceil, floatCeil, float, float)         \
+    F(F32Floor, floatFloor, float, float)       \
+    F(F32Trunc, floatTrunc, float, float)       \
+    F(F32Nearest, floatNearest, float, float)   \
+    F(F32Abs, floatAbs, float, float)           \
+    F(F32Neg, floatNeg, float, float)           \
+    F(I64Clz, clz, uint64_t, uint64_t)          \
+    F(I64Ctz, ctz, uint64_t, uint64_t)          \
+    F(I64Popcnt, popCount, uint64_t, uint64_t)  \
+    F(I64Eqz, intEqz, uint64_t, uint32_t)       \
+    F(F64Sqrt, floatSqrt, double, double)       \
+    F(F64Ceil, floatCeil, double, double)       \
+    F(F64Floor, floatFloor, double, double)     \
+    F(F64Trunc, floatTrunc, double, double)     \
+    F(F64Nearest, floatNearest, double, double) \
+    F(F64Abs, floatAbs, double, double)         \
+    F(F64Neg, floatNeg, double, double)
```

Comment:
It seems the 3rd and 4th arg is always the same. Is there an op where it will not be true?

Reply:
Currently there is no case that two arguments have different types.
But two macros `FOR_EACH_BYTECODE_BINARY_OP` and `FOR_EACH_BYTECODE_UNARY_OP` have same argument list so that its more convenient to use both macros at the same time.